src/engine/Engine.cpp

```diff
@@ -2838,10 +2838,14 @@ PiecewiseLinearConstraint *Engine::pickSplitPLConstraint( DivideStrategy strateg
         candidatePLConstraint = pickSplitPLConstraintBasedOnPolarity();
     else if ( strategy == DivideStrategy::EarliestReLU )
         candidatePLConstraint = pickSplitPLConstraintBasedOnTopology();
-    else if ( strategy == DivideStrategy::LargestInterval &&
-              ( _smtCore.getStackDepth() % GlobalConfiguration::INTERVAL_SPLITTING_FREQUENCY ==
+    else if ( !_produceUNSATProofs && strategy == DivideStrategy::LargestInterval &&
```

Comment:
If we turn this option off when producing proofs (as I agree we should), shouldn't we add !_produceUNSATProofs as a condition for choosing DivideStrategy::LargestInterval in the function Engine::decideBranchingHeuristics as well?

Reply:
That's a good point. Let me do that.

Reply:
UPDATE: actually, I should only turn input splitting off in decideBranchingHeuristics(), instead of during the actual solving.